python/lsst/daf/butler/_butler.py

```diff
@@ -2512,12 +2512,13 @@ def validateConfiguration(
         # a DatasetRef for each defined instrument
         datasetRefs = []
 
+        dimensions = self.dimensions.extract(("instrument",))
```

Comment:
It's possible this is a bad idea and I should move it into the loop. I don't want code to break that doesn't have instrument in the universe.

Reply:
I'm similarly ambivalent.  I suppose you could try/except and skip the whole block if instrument isn't in the dimensions.

Reply:
I rearranged it a bit and fixed the related problem a couple of lines above where we were querying for the instrument dimension without checking the universe first.